ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasCodeFormatter.cs

```diff
@@ -152,35 +158,64 @@ public static string FormatSelection(RdFantomasFormatSelectionArgs args)
 
       if (CurrentVersion >= FantomasProtocolConstants.Fantomas5Version)
         return FSharpAsync.StartAsTask(
-            FormatSelectionMethod.Invoke(null, new[]
-            {
-              args.FileName.EndsWith(".fsi"), // isSignature
-              args.Source,
-              range,
-              ConvertToFormatConfig(args.FormatConfig)
-            }) as dynamic, null, null) // FSharpAsync<Tuple<string, Range>>
+            CodeFormatterType.InvokeMember("FormatSelectionAsync",
+              BindingFlags.Static | BindingFlags.InvokeMethod | BindingFlags.Public, Type.DefaultBinder, null, new[]
+              {
+                args.FileName.EndsWith(".fsi"), // isSignature
+                args.Source,
+                range,
+                ConvertToFormatConfig(args.FormatConfig)
+              }) as dynamic, null, null) // FSharpAsync<Tuple<string, Range>>
           .Result.Item1.Replace("\r\n", args.NewLineText);
 
       return FSharpAsync.StartAsTask(
-          FormatSelectionMethod.Invoke(null, new[]
-          {
-            args.FileName, range,
-            SourceOriginConstructor.Invoke(null, new object[] { args.Source }),
-            ConvertToFormatConfig(args.FormatConfig),
-            CreateFSharpParsingOptions.Invoke(GetParsingOptions(args.ParsingOptions).ToArray()),
-            Checker
-          }) as FSharpAsync<string>, null, null)
+          CodeFormatterType.InvokeMember("FormatSelectionAsync",
+            BindingFlags.Static | BindingFlags.InvokeMethod | BindingFlags.Public, Type.DefaultBinder, null, new[]
+            {
+              args.FileName, range,
+              SourceOriginConstructor.Invoke(null, new object[] { args.Source }),
+              ConvertToFormatConfig(args.FormatConfig),
+              CreateFSharpParsingOptions.Invoke(GetParsingOptions(args.ParsingOptions).ToArray()),
+              Checker
+            }) as FSharpAsync<string>, null, null)
         .Result.Replace("\r\n", args.NewLineText);
     }
 
-    public static string FormatDocument(RdFantomasFormatDocumentArgs args) =>
-      FSharpAsync.StartAsTask(
-          FormatDocumentMethod.Invoke(null, GetFormatDocumentOptions(args)) as FSharpAsync<string>,
-          null, null)
-        .Result.Replace("\r\n", args.NewLineText);
+    public static RdFormatResult FormatDocument(RdFantomasFormatDocumentArgs args)
+    {
+      var formatDocumentOptions = GetFormatDocumentOptions(args);
+      var formatDocumentAsync = CodeFormatterType.InvokeMember("FormatDocumentAsync",
+        BindingFlags.Static | BindingFlags.InvokeMethod | BindingFlags.Public, Type.DefaultBinder, null,
+        formatDocumentOptions);
+      var formatResult = FSharpAsync.StartAsTask((dynamic)formatDocumentAsync, null, null).Result;
+
+      if (CurrentVersion < Version60)
+        return new RdFormatResult(formatResult.Replace("\r\n", args.NewLineText), null);
```

Comment:
Could you add a comment why this is needed, please?

Reply:
You should be able to pass in `lf` as part of the [configuration](https://fsprojects.github.io/fantomas/docs/end-users/Configuration.html#end_of_line) btw.
(And override whatever the user had)

Reply:
This is a relic of the past. We can try to rewrite it in a separate PR.
https://github.com/JetBrains/resharper-fsharp/blob/80620e2c051ab1a9d30367565ef60f760cdc1639/ReSharper.FSharp/src/FSharp.Psi.Features/src/Formatter/FSharpReformatCode.fs#L98

Reply:
I think a comment about the version check would stil be good here. With a link to a PR or a commit.